### PR TITLE
Expose PositionalError type instead of opaque Box<dyn Error>

### LINKED ISSUE #21: Expose PositionalResult instead of Box<dyn Error>
It would be nice if we could expose the PositionalResult type instead of an opaque error. 
The result would be a cleaner API that provides more information for the user.

### PR DESCRIPTION
**Pull request by denmuprima**
Added a new error variant to handle parsing errors.
Adjusted the necessary generated code and improved macro hygiene.
Adjusted tests.
Removed glob imports, partially to test that macros generate clean code (fully qualified imports).
Fixes [#21](https://github.com/primait/positional.rs/issues/21).

### CODE REVIEW

File: positional/tests/external_types.rs [hunk 1]
```diff
@@ -30,7 +30,7 @@ impl Data {
 fn simple() {
     let rows = vec![Data::new(
         "the name".to_string(),
-        NaiveDate::from_ymd(2022, 1, 1),
+        NaiveDate::from_ymd_opt(2022, 1, 1).unwrap(),
```
Comment (matteosister) [inline positional/tests/external_types.rs:33]: :+1:

File: positional/tests/name_clashes.rs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    #[allow(dead_code)]
+    fn from_positional_row(_: &str) -> Self {
+        unreachable!()
+    }
+
+    #[allow(dead_code)]
+    fn to_positional_row(self) -> String {
+        unreachable!()
+    }
+}
```
Comment (matteosister) [inline positional/tests/name_clashes.rs:29]: uh! I didn't think about that :exploding_head:

**Review by matteosister — APPROVED** [review]
LGTM!

File: positional/tests/name_clashes.rs (hunk 2 above)
Comment (denmuprima) [reply positional/tests/name_clashes.rs:29]: This is probably overkill; I just wanted to make sure that the generated code definitely calls the versions in the Trait impls generated by positioal :D

**Pull request merged**